### DISCUSSION TITLE
Using lights with LightMyBricks Powered Up adapter

### DISCUSSION THREAD

**Discussion by ashlarx** [pybricks/support#1879] [Q&A]
I am adding lights to my Orient Express train that is powered by Technic hub running PyBricks. 

The lights are from LightMyBricks and are powered by their Powered Up adapter: *https://www.lightmybricks.eu/collections/diy-powering-lego-lights/products/lego-powered-up-cable-power-functions-2-0*

In the official app, this adapter can be used in two ways:
1. on its own that with light control from the app
2. with a motor attached, the lights are on when the motor is running

In PyBricks code the second option works well, the lights are on when the attached motor is turned on. 
But when i am trying to use it as separate lights (Light(Port.A)), i get a message that lights are not found in the selected port. 

Does anyone know if there is a way to use this adapter for just controlling lights in the PyBricks?

**Comment by laurensvalk** [comment]
Each device is normally detected as a particular type. Depending on how that custom cable is internally wired, it might get detected as a light, a motor, or nothing at all.

Does it work when you use `DCMotor(Port.A)`?

**Reply by ashlarx** [reply]
Yes, the adapter works for controlling a motor and detects 'DCMotor(Port.A)'. This means then that it is detected as a motor and light control would not work. 

But since the hub probably does not care how much power the motor is using, then it would actually work to describe it as a motor for controlling the lights and plug the motor to a separate port. There are no options for lights that could not be given to a DCMotor.

In my case then the setup would be:
`train_motor_1 =  DCMotor(Port.A)`
`train_motor_2 = DCMotor(Port.B)`
`lights = DCMotor(Port.C)`

Thank you for the quick reply!

**Reply by laurensvalk** [reply]
Yes, setting `motor.dc(level)` to a level from 0 to 100 is exactly the same what `light.on(level)` would do.

Glad to help!
